core/search/disjunction.hpp

```diff
@@ -105,19 +246,39 @@ class basic_disjunction final : public doc_iterator_base, score_ctx {
   virtual bool next() override {
     next_iterator_impl(lhs_);
     next_iterator_impl(rhs_);
-    return !doc_limits::eof(doc_.value = std::min(lhs_.value(), rhs_.value()));
+    doc_.value = std::min(lhs_.value(), rhs_.value());
+    if (doc_limits::eof(doc_.value)) {
+      return false;
+    }
+    if /*constexpr*/ (std::is_same<Adapter, position_score_iterator_adapter<DocIterator>>::value) {
+      this->update_attribute_range(doc_.value, lhs_, rhs_);
+    }
```

Comment:
let's avoid having this logic as a part of next/seek call, use attribute+visitor instead